modules/lang-painless/build.gradle

```diff
@@ -88,6 +88,14 @@ dependencies {
   docImplementation project(':server')
   docImplementation project(':modules:lang-painless')
   docImplementation 'com.github.javaparser:javaparser-core:3.18.0'
+  if (isEclipse) {
+    /*
+     * Eclipse isn't quite "with it" enough to understand the different
+     * source sets. This adds the dependency to all source sets so it
+     * can compile the doc java files.
+     */
+    implementation 'com.github.javaparser:javaparser-core:3.18.0'
```

Comment:
Should this just be the only one? Or does this way not work in Intellij?

Reply:
No, it's configured correctly for intellij because it actually separates these classpaths.

Reply:
Ah, thank you. That makes sense.